jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildDockerMojo.java

```diff
@@ -50,21 +49,25 @@ public void execute() throws MojoExecutionException {
     }
 
     if (!new DockerClient().isDockerInstalled()) {
-      throw new MojoExecutionException(HELPFUL_SUGGESTIONS.forDockerNotInstalled());
+      throw new MojoExecutionException(
+          HelpfulSuggestions.forDockerNotInstalled(HELPFUL_SUGGESTIONS_PREFIX));
     }
 
     MavenJibLogger mavenJibLogger = new MavenJibLogger(getLog());
     MavenProjectProperties mavenProjectProperties =
         MavenProjectProperties.getForProject(getProject(), mavenJibLogger, getExtraDirectory());
 
     try {
+      MavenHelpfulSuggestionsBuilder mavenHelpfulSuggestionsBuilder =
+          new MavenHelpfulSuggestionsBuilder(HELPFUL_SUGGESTIONS_PREFIX, this);
+
       ImageReference targetImage =
           ConfigurationPropertyValidator.getGeneratedTargetDockerTag(
               getTargetImage(),
               mavenJibLogger,
               getProject().getName(),
               getProject().getVersion(),
-              HELPFUL_SUGGESTIONS);
+              mavenHelpfulSuggestionsBuilder.build());
```

Comment:
Seems odd to instantiate a builder twice but with different parameterizations?

Reply:
Agreed - I think this is a consequence of how `HelpfulSuggestions` was set up initially - it has too many different parameters injected for it while producing messages at different points in the code where certain parameters have not been resolved yet. This will definitely get a overhaul once the eventing system is in place. We'll also need to reduce the extensive injections into `HelpfulSuggestions` - filed #856.